### PR TITLE
Adding edge case tests for overlaps operators.

### PR DESCRIPTION
**Pull request by brynrhodes**
This adds a few edge case tests for the overlaps, overlaps before and overlaps after operators. These tests are related to the technical correction submitted to the specification here:

https://jira.hl7.org/browse/FHIR-46051

The author's guide is unclear about the intended interpretation of "overlaps before" and "overlaps after", whereas the reference and logical specification are both clear. These tests cover the edges of:

1) overlaps for equal intervals (true)
2) overlaps where the first interval starts at the end of the second (true)
3) overlaps before for equal intervals (false)
4) overlaps after for equal intervals (false)

### CODE REVIEW

**Comment by brynrhodes** [comment]
These new tests have been run successfully on the Java engine (see Cooking with CQL session 83)

File: tests/cql/CqlIntervalOperatorsTest.xml [hunk 1]
```diff
@@ -1016,6 +1016,14 @@
...
 			<output>true</output>
 		</test>
+		<test name="IntegerIntervalOverlapsTrue2">
+			<expression>Interval[4, 10] overlaps Interval[4, 10]</expression>
+			<output>true</output>
+		</test>
+		<test name="IntegerIntervalOverlapsTrue3">
+			<expression>Interval[10, 15] overlaps Interval[4, 10]</expression>
+			<output>true</output>
+		</test>
```
Comment (suyashkumar) [inline tests/cql/CqlIntervalOperatorsTest.xml:1026]: (non blocking thought): would it make sense to add a few tests here (in another PR) for intervals with exclusive boundaries for completeness? it essentially would test that folks are indeed implementing `overlaps` with the proper start/end semantics.

**Review by suyashkumar — APPROVED** [review]
Thanks Bryn!

File: tests/cql/CqlIntervalOperatorsTest.xml (hunk 1 above)
Comment (brynrhodes) [reply tests/cql/CqlIntervalOperatorsTest.xml:1026]: Yep, I think that makes sense

**Comment by antvaset** [comment]
The Java engine passes all new tests as Bryn mentioned, see https://github.com/cqframework/clinical_quality_language/pull/1530. That PR can be merged once this is approved and merged.

**Review by antvaset — APPROVED** [review]
